pandas/tests/arithmetic/conftest.py

```diff
@@ -174,44 +171,64 @@ def box_df_fail(request):
     return request.param
 
 
-@pytest.fixture(params=[(pd.Index, False),
-                        (pd.Series, False),
+@pytest.fixture(params=[pd.Index,
+                        pd.Series,
                         (pd.DataFrame, False),
-                        pytest.param((pd.DataFrame, True),
-                                     marks=pytest.mark.xfail(strict=True))],
+                        (pd.DataFrame, True),
+                        tm.to_array],
                 ids=id_func)
-def box_transpose_fail(request):
+def box_and_transpose(request):
     """
-    Fixture similar to `box` but testing both transpose cases for DataFrame,
-    with the tranpose=True case xfailed.
+    Like `box4`, but the DataFrame case with box transpose=True and
+    transpose=False
     """
     # GH#23620
     return request.param
 
 
-@pytest.fixture(params=[pd.Index, pd.Series, pd.DataFrame, PeriodArray],
+@pytest.fixture(params=[pd.Index,
+                        pd.Series,
+                        (pd.DataFrame, False),
+                        pytest.param((pd.DataFrame, True),
+                                     marks=pytest.mark.xfail(strict=True)),
+                        tm.to_array],
                 ids=id_func)
-def box_with_period(request):
+def box_transpose_fail(request):
     """
-    Like `box`, but specific to PeriodDtype for also testing PeriodArray
+    Like `box_and_transpose`, but xfailing the transposed DataFrame case
     """
+    # GH#23620
     return request.param
 
 
-@pytest.fixture(params=[pd.Index, pd.Series, pd.DataFrame, DatetimeArray],
-                ids=id_func)
-def box_with_datetime(request):
-    """
-    Like `box`, but specific to datetime64 for also testing DatetimeArray
-    """
-    return request.param
+# Construct tuples of the form (box, tz) over all our box classes and
+#  the timezones in tz_naive_fixture.  We then xfail the cases where the box
+#  is a DataFrame-with-transpose and the timezone is not tz-naive (i.e. None)
+boxes = [pd.Index,
+         pd.Series,
+         (pd.DataFrame, False),
+         (pd.DataFrame, True),
+         tm.to_array]
 
+# copied from pandas.conftest tz_naive_fixture
+TIMEZONES = [None, 'UTC', 'US/Eastern', 'Asia/Tokyo', 'dateutil/US/Pacific',
+             'dateutil/Asia/Singapore']
 
-@pytest.fixture(params=[pd.Index, pd.Series, pd.DataFrame, TimedeltaArray],
-                ids=id_func)
-def box_with_timedelta(request):
+params = [(x, y) for x in boxes for y in TIMEZONES]
+for n in range(len(params)):
+    tup = params[n]
+    if isinstance(tup[0], tuple) and tup[0][1] is True and tup[1] is not None:
+        # i.e. (DataFrame, True, tzinfo) excluding the no-tzinfo case
+        param = pytest.param(tup, marks=pytest.mark.xfail(strict=True))
+        params[n] = param
+
+
+@pytest.fixture(params=params)
```

Comment:
so my comment above was fairly general in nature. I don't think you need to test tz's other than [None, 'UTC'] or maybe use 'US/Eastern' rather than UTC to make sure that no times are shifted.

you actually just care if there IS a timezone or not.